pkg/kubelet/cm/cpumanager/cpu_manager.go

```diff
@@ -206,46 +211,57 @@ func (m *manager) Start(activePods ActivePodsFunc, sourcesReady config.SourcesRe
 	return nil
 }
 
-func (m *manager) AddContainer(p *v1.Pod, c *v1.Container, containerID string) error {
+func (m *manager) Allocate(p *v1.Pod, c *v1.Container) error {
 	m.Lock()
-	// Proactively remove CPUs from init containers that have already run.
-	// They are guaranteed to have run to completion before any other
-	// container is run.
-	for _, initContainer := range p.Spec.InitContainers {
-		if c.Name != initContainer.Name {
-			err := m.policyRemoveContainerByRef(string(p.UID), initContainer.Name)
-			if err != nil {
-				klog.Warningf("[cpumanager] unable to remove init container (pod: %s, container: %s, error: %v)", string(p.UID), initContainer.Name, err)
-			}
-		}
-	}
+	defer m.Unlock()
+
+	// TODO: Reuse CPUs allocated to init containers to mimic functionality
+	// from previous implementation below. Logic should probably be pushed into
+	// the policy now instead of doing it at this level.
+
+	//// Proactively remove CPUs from init containers that have already run.
+	//// They are guaranteed to have run to completion before any other
+	//// container is run.
+	//for _, initContainer := range p.Spec.InitContainers {
+	//	if c.Name != initContainer.Name {
+	//		err := m.policyRemoveContainerByRef(string(p.UID), initContainer.Name)
+	//		if err != nil {
+	//			klog.Warningf("[cpumanager] unable to remove init container (pod: %s, container: %s, error: %v)", string(p.UID), initContainer.Name, err)
+	//		}
+	//	}
+	//}
 
 	// Call down into the policy to assign this container CPUs if required.
-	err := m.policyAddContainer(p, c, containerID)
```

Comment:
Hmm, is the idea here to delay adding the container to the `containerMap` to the reconciliation loop? It seems that reconciliation loop might not get far enough in the loop to actually perform the Add().

Reply:
No, the it will be added to the containerMap as soon as it can be (i.e. at the top `AddContainer()` below).

Reply:
I asked because I only see `m.containerMap.Add()` being called in the reconciliation loop, but maybe it's abstracted somewhere...